build.gradle

```diff
@@ -41,6 +52,122 @@ subprojects {
             implementation files("$project.parent.buildDir/intermediates/javac/debug/compileDebugJavaWithJavac/classes/")
         }
 
+        task prepareDirs {
+            doLast {
+                mkdir "$projectDir.parent/subprojects"
+            }
+        }
+
+        tasks.each {
+            if (it != prepareDirs && it != clean) {
+                it.dependsOn prepareDirs
+            }
+        }
```

Comment:
IMO, It is better to add hook for ensuring subproject folder on one start (mostly task before build) rather than every task.
Please try with 
```
task ensureFolder {
            doLast {
                mkdir "$projectDir.parent/subprojects"
            }
        }
        compileJava.finalizedBy(ensureFolder)
```

Reply:
Modified the task.